source/common/stats/custom_stat_namespaces_impl.cc

```diff
@@ -0,0 +1,25 @@
+#include "source/common/stats/custom_stat_namespaces_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+bool CustomStatNamespacesImpl::registered(const absl::string_view name) const {
+  return namespaces_.find(name) != namespaces_.end();
+}
+
+void CustomStatNamespacesImpl::registerStatNamespace(const absl::string_view name) {
+  namespaces_.insert(std::string(name));
+};
+
+absl::optional<std::string>
+CustomStatNamespacesImpl::stripRegisteredPrefix(const absl::string_view stat_name) const {
+  const auto pos = stat_name.find_first_of('.');
+  if (pos != std::string::npos && registered(stat_name.substr(0, pos))) {
+    // Trim the custom namespace.
+    return std::string(stat_name.substr(pos + 1));
```

Comment:
You could return this as an `optional<string_view>`. This function doesn't need to copy the bytes in the string.